web-frontend/src/pages/Settings/SettingsUpdate.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useRef, SyntheticEvent, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { usePut } from '../../custom/api';
+import { AccountObject } from '../../models/AccountModel';
+import {
+  BAD_REQUEST,
+  NOT_FOUND,
+  UNPROCESSABLE_ENTITY,
+} from '../../models/status';
+import { routeFor } from '../../utils';
+import Intl from '../../i18n/i18n';
+import { isNicknameValid } from '../../validation/nickname';
+import { parseForm } from '../../utils/parseForm';
+
+import WalletLayout, {
+  MenuItem,
+} from '../../layouts/WalletLayout/WalletLayout';
+
+import { Button, Identicon, Input } from '@massalabs/react-ui-kit';
+import { SettingsOption } from './SettingsOption';
+import { AxiosError } from 'axios';
+
+interface IErrorObject {
+  code?: string;
+  message?: string;
+}
+
+export default function SettingsUpdate() {
+  const navigate = useNavigate();
+  const { nickname } = useParams();
+  const form = useRef(null);
+
+  const [newNickname, setNewNickname] = useState<string>(nickname ?? '');
+  const [error, setError] = useState<IErrorObject | null>(null);
+
+  const {
+    mutate: mutableUpdate,
+    isSuccess: isSuccessUpdate,
+    error: errorUpdate,
+  } = usePut<AccountObject>(`accounts/${nickname}`);
+
+  function handleErrors() {
+    const err: AxiosError = errorUpdate as AxiosError;
+
+    switch (err.response?.status) {
+      case BAD_REQUEST:
+        setError({ message: Intl.t('errors.account-invalid-format') });
+        break;
+      case NOT_FOUND:
+        setError({ message: Intl.t('errors.account-not-found') });
+        break;
+      case UNPROCESSABLE_ENTITY:
+        setError({ message: Intl.t('errors.account-already-exists') });
+        break;
+      default:
+        setError({ message: Intl.t('errors.unexpected-error.description') });
+        break;
+    }
+  }
```

Comment:
Humm, you can't rely all BAD_REQUEST, and/or NOT_FOUND and etc to map to these errors. A BAD_REQUEST and etc can be more things than only invalid format.

Reply:
> Humm, you can't rely all BAD_REQUEST, and/or NOT_FOUND and etc to map to these errors. A BAD_REQUEST and etc can be more things than only invalid format.

SO what can I do ? 🤔

Reply:
Corrected!
But we have to check if backend return good code (500, 422, etc...) in another PR.